arch/arm64/src/imx9/imx9_usbdev.c

```diff
@@ -776,8 +784,18 @@ static inline void imx9_writedtd(struct imx9_dtd_s *dtd,
 
   up_clean_dcache((uintptr_t)dtd,
                   (uintptr_t)dtd + sizeof(struct imx9_dtd_s));
-  up_clean_dcache((uintptr_t)data,
-                  (uintptr_t)data + nbytes);
+
+  /* the pointer is NULL for EP0 NAK IN/OUT */
+
+  if (data != NULL)
+    {
+      DEBUGASSERT(IS_CACHE_ALIGNED(data, CACHE_ALIGN_UP(nbytes)) &&
```

Comment:
Suggest splitting the DEBUGASSERT into two because it makes it easier to see which test is failing / causing the assert (from line number)

Reply:
sure